src/entities/createPool/createPoolV3/weighted/createPoolWeighted.ts

```diff
@@ -21,30 +20,16 @@ export class CreatePoolWeightedV3 implements CreatePoolBase {
     }
 
     private encodeCall(input: CreatePoolV3WeightedInput): Hex {
-        const sortedTokenParams = sortByAddress(input.tokens);
+        const sortedTokenConfigs = sortByAddress(input.tokens);
 
-        const [tokenConfigs, normalizedWeights] = sortedTokenParams.reduce(
-            (
-                acc,
-                {
-                    address: tokenAddress,
-                    rateProvider,
-                    weight,
-                    tokenType,
-                    paysYieldFees,
-                },
-            ) => {
-                acc[0].push({
-                    token: tokenAddress,
-                    tokenType,
-                    rateProvider,
-                    paysYieldFees: paysYieldFees ?? false,
-                });
-                acc[1].push(weight);
-                return acc;
-            },
-            [[], []] as [TokenConfig[], bigint[]],
-        );
+        const normalizedWeights: bigint[] = [];
+        const tokens = sortedTokenConfigs.map(({ weight, ...rest }) => {
+            normalizedWeights.push(weight);
+            return {
+                token: rest.address,
+                ...rest,
+            };
+        });
```

Comment:
This was only place `TokenConfig` type used. I hijacked the type for create pool inputs. 

Ideally, we wouldn't have to map over the `sortedTokenConfigs` to change prop name from `address` to `token` but the refactor is fairly involved

Reply:
You are saying that for example:
```
export type CreatePoolV3WeightedInput = CreatePoolV3BaseInput & {
    poolType: PoolType.Weighted;
    tokens: {
        address: Address;
        rateProvider: Address;
        weight: bigint;
        tokenType: TokenType;
        paysYieldFees?: boolean;
    }[];
};
```
The `tokens` property should ideally already be `TokenConfig[]` instead of `[]`?

Reply:
ya i think it makes sense to re-use across v3 creation inputs 

https://github.com/balancer/b-sdk/blob/9d4e09766a760038117fedc94e1c24aab1a37f5d/src/entities/createPool/types.ts#L66-L69

Reply:
I like this refactor! Maybe in the future it would be useful to have a bit of a clearer description of the changes and reasoning in the main description, would potentially make it easier to follow when reviewing.

Reply:
You have no idea how happy I am that you're removing these `reduce` from the code 😅 
Thanks a LOT for this refactor! 👏